src/shared/hooks/useInterval.ts

```diff
@@ -1,7 +1,7 @@
 import { useEffect, useRef } from 'react';
 
-function useInterval(callback, delay) {
-	const savedCallback = useRef();
+function useInterval(callback: (...args: any[]) => void, delay: number | null) {
+	const savedCallback = useRef(callback);
```

Comment:
not sure I follow... `callback` is should just be `() => void` no?
also, why is the callback the initial value of the `useRef`? this wasn't in the original implementation

Reply:
ah typescript is complaining, wtf - anyways, found this - https://github.com/Hermanya/use-interval/blob/master/src/index.tsx - you could just use `() => {}` inside and this would make TS happy.

Reply:
I can change the callback although I was basically saying it can take any number of arguments

Reply:
right... but its not really used anywhere (the args of the callback), could you replace `useRef(callback)` with something like `useRef(() => {})` to make TS happy? (as shown in the implementation above)